pkg/auth/handler/api/anonymous_user_signup.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/lib/infra/db/appdb"
+	oauthhandler "github.com/authgear/authgear-server/pkg/lib/oauth/handler"
+	"github.com/authgear/authgear-server/pkg/util/httproute"
+	"github.com/authgear/authgear-server/pkg/util/httputil"
+	"github.com/authgear/authgear-server/pkg/util/log"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func ConfigureAnonymousUserSignupRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("POST", "OPTIONS").
+		WithPathPattern("/api/anonymous_user/signup")
+}
+
+var AnonymousUserSignupAPIRequestSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false,
+		"properties": {
+			"client_id": { "type": "string" },
+			"session_type": {
+				"type": "string",
+				"enum": ["cookie", "refresh_token"]
+			}
+		},
+		"required": ["client_id", "session_type"]
+	}
+`)
+
+var AnonymousUserSignupAPIResponseSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"properties": {
+			"token_type": { "type": "string" },
+			"access_token": { "type": "string" },
+			"refresh_token": { "type": "string" },
+			"expires_in": { "type": "string" }
+		},
+		"required": ["token_type", "access_token", "refresh_token", "expires_in"]
+	}
+`)
+
+type AnonymousUserSignupAPIRequest struct {
+	ClientID    string                      `json:"client_id"`
+	SessionType oauthhandler.WebSessionType `json:"session_type"`
+}
+
+func (p *AnonymousUserSignupAPIRequest) Validate(ctx *validation.Context) {
+	if !p.SessionType.IsValid() {
+		ctx.Child("session_type").EmitErrorMessage("unknown session type")
+	}
+}
+
+type JSONResponseWriter interface {
+	WriteResponse(rw http.ResponseWriter, resp *api.Response)
```

Comment:
If we use JSONResponseWriter to write response, the response will be wrapped under the `result` key, which does NOT match the declared response schema.

Reply:
This is intended, since we had a convention for the api response, the success result should has key `result`. Updated the general API spec to mention it. see https://github.com/authgear/authgear-server/pull/1708